connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -497,12 +506,18 @@ private SinkRecord convertAndTransformRecord(final ConsumerRecord<byte[], byte[]
                 timestamp,
                 msg.timestampType(),
                 headers);
+
+        InternalSinkRecord internalSinkRecord = origRecord.newRecord(origRecord.topic(),
+            origRecord.kafkaPartition(), origRecord.keySchema(), origRecord.key(),
+            origRecord.valueSchema(), origRecord.value(), origRecord.kafkaOffset(),
+            origRecord.timestamp(), origRecord.timestampType(), origRecord.headers(), msg);
+
         log.trace("{} Applying transformations to record in topic '{}' partition {} at offset {} and timestamp {} with key {} and value {}",
                 this, msg.topic(), msg.partition(), msg.offset(), timestamp, keyAndSchema.value(), valueAndSchema.value());
         if (isTopicTrackingEnabled) {
-            recordActiveTopic(origRecord.topic());
+            recordActiveTopic(internalSinkRecord.topic());
```

Comment:
This line would not need to be affected.
```suggestion
            recordActiveTopic(sinkRecord.topic());
```